pulsar-broker/src/test/java/org/apache/pulsar/broker/service/plugin/FilterEntryTest.java

```diff
@@ -463,4 +472,93 @@ private void verifyBacklog(String topic, String subscription,
         Assert.assertEquals(numMessagesRejected, a1.getFilterRejectedMessages());
         Assert.assertEquals(numMessagesRescheduled, a1.getFilterRescheduledMessages());
     }
+
+
+    @DataProvider(name = "overrideBrokerEntryFilters")
+    public static Object[][] overrideBrokerEntryFilters() {
+        return new Object[][]{ {true}, {false} };
+    }
+
+
+    @Test(dataProvider = "overrideBrokerEntryFilters")
+    public void testExecuteInOrder(boolean overrideBrokerEntryFilters) throws Exception {
+        conf.setAllowOverrideEntryFilters(true);
+        String topic = "persistent://prop/ns-abc/topic" + UUID.randomUUID();
+        String subName = "sub";
+        Producer<String> producer = pulsarClient.newProducer(Schema.STRING)
+                .enableBatching(false).topic(topic).create();
+        for (int i = 0; i < 10; i++) {
+            producer.send("test");
+        }
+
+        EntryFilterWithClassLoader mockFilterReject = mock(EntryFilterWithClassLoader.class);
+        when(mockFilterReject.filterEntry(any(Entry.class), any(FilterContext.class))).thenReturn(
+                EntryFilter.FilterResult.REJECT);
+        EntryFilterWithClassLoader mockFilterAccept = mock(EntryFilterWithClassLoader.class);
+        when(mockFilterAccept.filterEntry(any(Entry.class), any(FilterContext.class))).thenReturn(
+                EntryFilter.FilterResult.ACCEPT);
+        if (overrideBrokerEntryFilters) {
+            setMockFilterToTopic((PersistentTopic) pulsar.getBrokerService()
+                    .getTopicReference(topic).get(), List.of(mockFilterReject, mockFilterAccept));
+        } else {
+            setMockFilterToTopic((PersistentTopic) pulsar.getBrokerService()
+                    .getTopicReference(topic).get(), List.of());
+            setMockBrokerFilter(List.of(mockFilterReject, mockFilterAccept));
+        }
```

Comment:
I think that we need to get rid of this type of mocking. It's not that this PR has introduced this approach. 

It would be useful it the EntryFilterProvider could be overridden in tests. It becomes easier if EntryFilterProvider is a simple interface and the existing EntryFilterProvider is renamed to EntryFilterProviderImpl or DefaultEntryFilterProvider. 

Instead of injecting fields and doing other hacks, filters could simply be tested by passing an EntryFilterProvider in tests.